Kryptonite/Silo.swift

```diff
@@ -305,6 +304,32 @@ class Silo {
             }
             
             responseType = .me(MeResponse(me: MeResponse.Me(email: try keyManager.getMe(), publicKeyWire: try keyManager.keyPair.publicKey.wireFormat(), pgpPublicKey: pgpPublicKey)))
+            
+        case .hosts:
+            
+            if allowed {
+                // git: get the list of PGP user ids
+                let pgpUserIDs = try KeyManager.sharedInstance().getPGPUserIDList()
+                
+                // ssh: read the logs and get a unique set of user@hostnames
+                let sshLogs:[SSHSignatureLog] = LogManager.shared.fetchAll()
```

Comment:
is there a chance this runs out of memory? maybe there is a way to stream signature logs from coredata and collect the unique user/host combinations incrementally?

Reply:
@kcking  addressed this in the newest commit, please take a look